data_structure.py

```diff
@@ -757,6 +758,22 @@ def updateNode(self, context):
 ##############################################################
 ##############################################################
 
+@contextmanager
+def context_sensitive_throttle(node_tree):
+    if node_tree.skip_tree_update:
```

Comment:
Uh. Sorry, got to this only now.
I do not see the reason for this check. Why do not just `node_tree.skip_tree_update = True`? If it's already True — that's will do nothing, but no problem?

Reply:
if its already `True`, then another function call is already throttling it, and it won't do anything but yield `node_tree`, but it must be sure not to set it to `False` .

Reply:
Ah, I see. Then it's probably better to do
```
prev_skip = node_tree.skip_tree_update
node_tree.skip_tree_update = True
...
node_tree.skip_tree_update = prev_skip
```
(plus usual try-catch stuff, off course).

Reply:
all we want there is a `no-op` , because the scenario is that there's a throttle, inside a throttle, (a nested throttle), and we dont want to prematurely 'exit' the throttle state.

exactly what code is used for this is really down to what's most effective. i welcome your suggestion.

Reply:
```
        try:
            prev_skip_state = node_tree.skip_tree_update
            node_tree.skip_tree_update = True
            yield node_tree
        finally:
            node_tree.skip_tree_update = prev_skip_state
```
this approach has the following properties:
1) it guarantees that skip_tree_update will be returned to it's previous state, no matter what it was;
2) it will automatically work with any nesting level, because prev_skip_state will be stored as a local variable in the call stack;
3) the fact that we are just restoring the state that was before is obvious from the code, it does not raise questions like "what if state was True? Or what if it was None for some reason — will it be restored afterwards?"
4) it does not require two copies of try-finally clause — "what if true" and "what if false".

Reply:
ok :) yes!